DEVELOPMENT.md

```diff
@@ -82,6 +82,7 @@ is the easiest. Just make sure to download the right version. It'll manage thing
 useful menulet where the server can be turned on and off. Don't forget to create a server if you
 go this route.
 
+In order to run tests using a local Postgres instance, you'll need to locally modify the values in [`application-test.yml`](https://github.com/DataBiosphere/terra-workspace-manager/blob/main/service/src/test/resources/application-test.yml)
```

Comment:
Good point - modified this a bit so you can now set `TEST_LOCAL_DB` to point to a local server and have tests run with a local DB.

One issue is that running parallel tests against a single shared DB causes interference between tests (DirtiesContext, cleaning DB on start, etc). I added a second env var for this (`TEST_SINGLE_THREAD`) because that felt like a weird side-effect to tack onto having `TEST_LOCAL_DB` defined, but either way would work. Do you have strong opinions on one env var vs. 2?

Reply:
I think two envvars is just fine, thanks! Seems cleaner to me to have them separate, FWIW

Reply:
Interesting... this interference doesn't happen on GHA because each test task uses a separate docker container?

The cleaning DB on start -- that happens once per test task (unit, connected, ...)?

Doesn't matter to me if I have to set 1 or 2 env vars. When I'm working on a test, I run just that test class. If I do run all connected tests, I would like to run in parallel.

Reply:
Yep! With TestContainers each thread is talking to its own DB container so they don't interfere with eachother.

Each thread will run the liquibase cleanup + intialization once on startup, so with multiple threads it happens multiple times (often after other threads have already started running tests, plus additional wipes for tests with `@DirtiesContext`)

Reply:
Just curious, if there wasn't TEST_SINGLE_THREAD workaround, for local tests would I have interference problems:

1. Run all connected tests
2. Run connected and unit at same time

I assume yes for 2. Just curious if problem would happen for 1.

Reply:
If you're using a local database with `TEST_LOCAL_DB` but without `TEST_SINGLE_THREAD`, both 1 and 2 would have interference problems - the Gradle runners don't all start at exactly the same time and each one will wipe the DB on start, and there are still `@DirtiesContext` connected tests.

If you're using TestContainers locally (i.e. `TEST_LOCAL_DB` is not set), the tests will run in parallel and 1 and 2 will both work since each runner has its own DB container.